nullaway/src/main/java/com/uber/nullaway/fixserialization/out/EnclosingNode.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2022 Uber Technologies, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.uber.nullaway.fixserialization.out;
+
+import com.google.errorprone.util.ASTHelpers;
+import com.sun.source.tree.ClassTree;
+import com.sun.source.tree.MethodTree;
+import com.sun.source.tree.Tree;
+import com.sun.source.util.TreePath;
+
+/** Nodes with this type, are surrounded in a method in source code. */
+public abstract class EnclosingNode {
```

Comment:
I think rather than having this class as abstract and having `ErrorInfo` and `SuggestedFixInfo` inherit from it, it should be a separate class `EnclosingClassAndMethodInfo`, and each of `ErrorInfo` and `SuggestedFixInfo` should have a field of type `EnclosingClassAndMethodInfo`.  `ErrorInfo` and `SuggestedFixInfo` can still have a method `initEnclosing()` that delegates to the `EnclosingClassAndMethodInfo` object.  What do you think?

Reply:
Hi @msridhar, this sounds cleaner to me, thank you for the suggestion. Inheritance relations here  seems unnecessary. I was inspired by how ```javaparser``` is written where there is a lot of these types such as ```NodeWithX``` which I don't think it is applicable here. 

c7398ca99ef264b857309cea1f7040bfd2f71926